src/Engine/ProtoCore/Reflection/MemberInfoReflectionExtensions.cs

```diff
@@ -0,0 +1,211 @@
+                        {
+                            var valueColl = item.Value as ICollection;
+                            var argsArray = new object[valueColl.Count];
+                            var idx = 0;
+                            foreach (var e in item.Value as ICollection)
+                            {
+                                if (e is CustomAttributeTypedArgument cs)
+                                    argsArray[idx] = cs.Value.InstanceFromObject(cs.ArgumentType);
+                                else
+                                    argsArray[idx] = e;
+                                idx++;
+                            }
+                            if (TryConvertToTypeArray(argsArray, out Array typeArray))
+                            {
+                                paramValues.Add(typeArray);
+                                continue;
+                            }
+                            paramValues.Add(argsArray);
+                            continue;
+                        }
+
+                        if (item.Value is ICollection enumerable)
+                        {
+                            foreach (var e in enumerable)
+                            {
+                                if (e is CustomAttributeTypedArgument cs)
+                                    paramValues.Add(cs.Value.InstanceFromObject(cs.ArgumentType));
+                                else
+                                    paramValues.Add(e);
+                            }
+
+                            continue;
+                        }
+                        paramValues.Add(item.Value.InstanceFromObject(item.ArgumentType));
+                    }
+
+
+                    dynamic ctorParam = null;
+                    if (paramValues.Count > 1)
+                    {
+                        var array = paramValues.ToArray<object>();
+                        if (TryConvertToTypeArray(array, out Array newArray))
+                        {
+                            ctorParam = newArray;
+                        }
+
+                        else
+                        {
+                            ctorParam = array;
+                        }
+                    }
+                    else
+                    {
+                        ctorParam = paramValues.FirstOrDefault();
+                    }  
+
+                    var a = CreateInstance(type, ctorParam);
+                    if (a is null) continue;
+
+                    attributes.Add(a);
+                }
+                catch (Exception e)
+                {
+                }
+            }
+
+            return attributes.ToArray();
+        }
+
+        private static Attribute CreateInstance(System.Type type, object args)
+        {
+            var loadedAssembly = Assembly.LoadFrom(type.Assembly.Location);
+            // Check if in right assembly before laoding...
+            var loadedType = loadedAssembly.GetType(type.FullName);
```

Comment:
Might no be understanding this lines, but why cannot this use the input `Type` directly instead of loading its assembly (which should be loaded already as you have its type as input?) to get the type again?

Reply:
the type will in this case only be loaded into reflection context (or the MetadataLoadContext as im using for this), when this is the case you cannot create instances from it, so i need to actually load it before its possible to create instances from the type. We are only using this to get Dynamo defined attributes so we know that the assemblies that defines these types are available to us and there for we can load the assembly and the get the type from the Loaded assembly instead of the MLC assembly

Reply:
This will force the assembly to be loaded then. Don't have much context to say but if this is the desired behaviour then it's fine. Been recently pulling my hair dealing with assemblies and app domains and is not pretty